### DISCUSSION TITLE
How to create a separated dialog like Alipay's Mini Program Studio?

### DISCUSSION THREAD

**Discussion by mister-teddy** [opensumi/core#1331] [Q&A] [answered]
We are developing another Mini Program Studio, and Alipay is our ideal result. When you first open the Studio, a separated window (which is not VS Code) popup to let you login and create new projects,... I wonder if there is any existing ContributionPoint or Component that I can easily extends to implement this feature? Or I would need to edit the electron main code to create another BrowserWindow (in the case how could I use the DI system and React)?

<img width="1264" alt="Screen Shot 2022-07-12 at 19 07 19" src="https://user-images.githubusercontent.com/29925961/178486181-d5f8391e-3f0c-46b3-a404-401ee5f54bfa.png">

Thank you very much for your support!

**Comment by bytemain** [comment]
Yes, you need write some code in the electron main process to control the welcome page.

first of all, let's see this code example: https://github.com/opensumi/ide-electron/blob/main/src/main/launch.ts#L49)

<img width="749" alt="CleanShot 2022-07-12 at 20 47 14@2x" src="https://user-images.githubusercontent.com/13938334/178493304-67f29025-a4cc-453c-83c5-9cc5e61afd76.png">

inside this class, it will initialize the di system, so you can use this injector

I can use a pseudo code to explain:

```ts

class MyAlipayMiniProgramStudio {
  start() {
    const app = new ElectronMainApp(xxxxxx);
    this.injector = app.injector;
    this.webcomeWindow = this.injector.get(WelcomeWindow)
    this.welcomeWindow.show()
  }
}

```

As for the content of this independent window, just like any other normal electron app,

create a BrowserWindow, then loadURL, ...

**Reply by mister-teddy** [reply]
Thank you for your snippet 🙏

**Comment by yantze** [comment, marked as answer]
A project similar to this has been opened https://github.com/opensumi/app-desktop

https://user-images.githubusercontent.com/2226423/178495111-5e5f5844-ad0d-4f17-95c6-4ee58aa46c0f.mov

**Reply by mister-teddy** [reply]
Thank you very much! If only I know about this project earlier 😅
